### PR TITLE
feat/Python: Regular tx fields

### LINKED ISSUE #917: Adapt fields of RegularTransactionEssence - python
Blocked by https://github.com/iotaledger/iota-sdk/issues/911 and https://github.com/iotaledger/iota-sdk/issues/915

https://github.com/iotaledger/tips-draft/blob/tip45/tips/TIP-0045/tip-0045.md#serialized-layout

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

blocked by https://github.com/iotaledger/iota-sdk/issues/915 and https://github.com/iotaledger/iota-sdk/issues/911

## Links to any relevant issues

Fixes [#917](https://github.com/iotaledger/iota-sdk/issues/917)

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
This can be undrafted now ?

File: bindings/python/iota_sdk/types/essence.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        allotments: The allotments of Mana which which will be added upon commitment of the slot.
+        payload: An optional tagged data payload
+    """
+    network_id: str
+    # TODO: Change to SlotIndex
+    creation_slot: HexStr
+    inputs: List[UtxoInput]
+    inputs_commitment: HexStr
+    outputs: List[BasicOutput | AccountOutput | FoundryOutput | NftOutput | DelegationOutput]
+    context_inputs: Optional[List[CommitmentInput | BlockIssuanceCreditInput | RewardInput]] = None
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/essence.py]: Should use the Union type for Python 3.9 compatibility

Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/essence.py]: same order as the doc please

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/python/iota_sdk/types/essence.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        allotments: The allotments of Mana which which will be added upon commitment of the slot.
+        payload: An optional tagged data payload
+    """
+    network_id: str
+    # TODO: Replace with a proper SlotIndex type
+    creation_slot: HexStr
+    context_inputs: Optional[List[Union[CommitmentInput | BlockIssuanceCreditInput | RewardInput]]] = None
+    inputs: List[UtxoInput]
+    inputs_commitment: HexStr
+    outputs: List[Union[BasicOutput | AccountOutput | FoundryOutput | NftOutput | DelegationOutput]]
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/essence.py]:
```suggestion
    context_inputs: Optional[List[Union[CommitmentInput, BlockIssuanceCreditInput, RewardInput]]] = None
    inputs: List[UtxoInput]
    inputs_commitment: HexStr
    outputs: List[Union[BasicOutput, AccountOutput, FoundryOutput, NftOutput, DelegationOutput]]
```

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**